src/disk_io_thread.cpp

```diff
@@ -1074,9 +1088,12 @@ namespace libtorrent
 			// don't evict write jobs if at least one other thread
 			// is flushing right now. Doing so could result in
 			// unnecessary flushing of the wrong pieces
-			if (evict > 0 && m_stats_counters[counters::num_writing_threads] == 0)
```

Comment:
I realize that this is a race condition. It it was a compare and swap, and we would flush "optimistically" (i.e. only if nobody else is flushing right now)